### ISSUE TITLE
PopupService Using Components -> Passing Component Instance - Not Component Class

### ISSUE THREAD

**Issue by meriturva** [telerik/kendo-angular#733]
We are starting to migrate our old approach based on kendo JQuery UI ContextMenu to new Popup / PopupServices approach (big improvements here, well done).

We just would like to use PopupService with Component because we need to keep DOM really clean (for performance and quality code) so we defined simple Component with a template to pass to PopupService.  But now we have big problem.

Our popup Component is provided by different module and it has few properties to set before being showed by PopupService.

Basically main question is: **how to show a Component Instance using PopupService (so passing instance not class) ?**

Just for telerik reference TID 1118917

Thanks.

**Comment by rusev** [comment]
We haven't consider this scenario in current implementation. The Popup expects a component type and it will perform the mechanic behind the dynamic instantiation of that component.

If setting properties on the content component is the only use-case you should be able to do this by using the [content](http://www.telerik.com/kendo-angular-ui/components/popup/api/PopupRef/#toc-content) property of the PopupRef.

**Comment by meriturva** [comment]
As i said i'm using Popup as a context menu (just to remove old dependency of Kendo JQuery and keep DOM tidy).
So i have a layout component where i put popupWrapper and then i have so many components that generate new "content" for popup before open event. I'm investigate a way to inject a component viewRef into templateRef before it is showed.

**Comment by ggkrustev** [comment]
Hi @meriturva,

I would like to second the @rusev's statement that PopupService expects either a TemplateRef or Component type. It cannot append/render a component instance. We've investigated further the case, but for the time being we do not have any plans to add built-in support for such functionality.

Right now, you can implement the desired behavior using a custom component that will handle the 
child content swaps. Once this is done, you can render it into the Popup component. 
IMHO this is more accurate, as the main Popup task is to position its content and nothing more.

If you believe that such functionality will be useful to others, please share your thoughts in our UserVoice portal.

http://kendoui-feedback.telerik.com/forums/555517-kendo-ui-for-angular-feedback?query=scheduler

I will close the issue, as we cannot do much for the time being.
